src/ngraph/runtime/cpu/pass/cpu_layout.cpp

```diff
@@ -1093,7 +1099,9 @@ namespace ngraph
                 void CPULayout::LAYOUT_DECL(ngraph::op::Reshape)
                 {
                     auto reshape = static_cast<ngraph::op::Reshape*>(node.get());
-                    if (reshape->get_is_transpose())
+                    if (reshape->get_is_transpose() &&
+                        reshape->get_output_shape().size() ==
+                            reshape->get_argument(0)->get_shape().size())
```

Comment:
why would **logical** sizes of input and output be different?

Reply:
Reshape could both transpose and collapse/change shape.

Reply:
Ah, those are *ranks*. I'm sorry; my bad 😄 I mistook them for `shape_size(get_shape())`